Corpus/residential.py

```diff
@@ -364,11 +364,26 @@ def receptacles(self):
                 eq = Equipment(**set_appliances[app])
                 r_app = dict()
                 n_app = 0
-                # loop for all household mmembers
+                # loop for all household members
                 for i in counter:
-                    r_appi, n_appi = eq.simulate(nday, dow, self.clusters[i], self.occ[i])
-                    r_app = stats.sum_dict(r_app, r_appi)
-                    n_app += n_appi
+                    # if appliances are not shared the load of using the appliance can be summed for each householdmember
+                    if app in ('CordlessPhone','PC'): # it is assumed that only the cordlessphone and PC are not shared. Each person in the household (>12) has its own phone or PC.
+                        r_appi, n_appi = eq.simulate(nday, dow, self.clusters[i], self.occ[i])
+                        r_app = stats.sum_dict(r_app, r_appi)
+                        n_app += n_appi
+                    else:
+                        # if appliances are shared the load of using the appliance can be maximal the cycle load. 
+                        # if the appliance is an appliance that is working continously the load is simulated for only 1 person
+                        if app in ('FridgeFreezer', 'Refrigerator', 'ChestFreezer', 'UprightFreezer', 'Clock'):
+                            if i == 0:
+                                r_appi, n_appi = eq.simulate(nday, dow, self.clusters[i], self.occ[i])
+                                r_app = stats.sum_dict(r_app, r_appi)
+                                n_app += n_appi  
+                        # if the appliance is not continously used, but sometimes shared, the cycle load is assigned when minimal one person is using it, when nobody is using it the stanby load is assigned
+                        else:
+                            r_appi, n_appi = eq.simulate(nday, dow, self.clusters[i], self.occ[i])
+                            r_app = stats.sum_dict2(r_app, r_appi)
+                            n_app += n_appi
```

Comment:
Actually, this seems to count the cycles for all occupants, even though the power has been taken as the max. This means the calibration will not be correct, as it uses this number of cycles. I don't see a quick fix... I'm thinking of a solution to perhaps pass on to the `simulate` function an aggregated occupancy & adjusted clusters, or make a new type of `stochastic_shared_load` in `simulate`, that will do this in there. Any suggestions welcome, otherwise I'll try to propose something soon.

Reply:
yes I see what's the problem. It is indeed difficult to solve in this part of the code. or in the way I tackled the sharing issue. (i did not look at the n_app). The problem is  that if two people share an appliance the total number of cycles will not be the same as the sum of the cycles performed by both people, but will also not be the number of cycles performed by only one person. I think indeed that a different function in simulate is needed as you suggest. In my previous code I tackled the sharing completely different. I made, just like you suggest, a new function in the simulate part. In this function I first checked if an occupant was using the appliance without assigning a load. So 0 if nobody is using it and then 1,2,3,... depending on how many people are using the appliance. In a next step I added a load depending on the type of appliance. So for shared if it was >0 the load was assigned, otherwise the standby-load was assigned. For non shared appliances the number was multiplied by the load. However, by doing it in this way the simulation time increased quite a lot.